src/features/country/countrySlice.js

```diff
@@ -0,0 +1,151 @@
+        newFilteredCountryName = state.countriesInfo.filter((country) =>
+          country.name.toLowerCase().includes(action.payload.toLowerCase())
+        );
+      }
+      state.filteredCountry = newFilteredCountryName;
+    },
+    getCurrentCountryIndex: (state, action) => {
+      const getIndex = () => {
+        return state.countriesInfo.findIndex(
+          (country) => country.name === state.countryName.slice(7)
+        );
+      };
+      const index = parseInt(getIndex(action.payload));
+      state.currentCountryIndexToNumber = index;
+    },
+    getMoreOptions: (state, action) => {
+      const lastCountryOfList = state.countriesInfo[249];
+      const secondLastCountryOfList = state.countriesInfo[248];
+      const firstCountryOfList = state.countriesInfo[0];
+      const secondCountryOfList = state.countriesInfo[1];
+      const firstBehindCurrentCountry =
+        state.countriesInfo[state.currentCountryIndexToNumber + 1];
+      const secondBehindCurrentCountry =
+        state.countriesInfo[state.currentCountryIndexToNumber + 2];
+      const twoAboveCurrentCountry =
+        state.countriesInfo[state.currentCountryIndexToNumber - 2];
+      const oneAboveCurrentCountry =
+        state.countriesInfo[state.currentCountryIndexToNumber - 1];
+
+      state.loadingMoreCountries = state.filteredCountry;
+
+      const unshift = (secondBeforeCurrentCountry, firstBeforeCurrentCountry) =>
+        state.filteredCountry.unshift(
+          secondBeforeCurrentCountry,
+          firstBeforeCurrentCountry
+        );
+      const push = (firstBehibdCurrentCountry, secondBehibdCurrentCountry) =>
+        state.filteredCountry.push(
+          firstBehibdCurrentCountry,
+          secondBehibdCurrentCountry
+        );
+      if (state.currentCountryIndexToNumber === 0) {
+        unshift(secondLastCountryOfList, lastCountryOfList);
+        push(firstBehindCurrentCountry, secondBehindCurrentCountry);
+
+        state.loadingMoreCountries = state.filteredCountry;
+      } else if (state.currentCountryIndexToNumber === 1) {
+        unshift(lastCountryOfList, oneAboveCurrentCountry);
+        push(firstBehindCurrentCountry, secondBehindCurrentCountry);
+
+        state.loadingMoreCountries = state.filteredCountry;
+      } else if (state.currentCountryIndexToNumber === 249) {
+        unshift(twoAboveCurrentCountry, oneAboveCurrentCountry);
+        push(firstCountryOfList, secondCountryOfList);
+
+        state.loadingMoreCountries = state.filteredCountry;
+      } else if (state.currentCountryIndexToNumber === 248) {
+        unshift(twoAboveCurrentCountry, oneAboveCurrentCountry);
+        push(firstBehindCurrentCountry, firstCountryOfList);
+
+        state.loadingMoreCountries = state.filteredCountry;
+      } else {
+        unshift(twoAboveCurrentCountry, oneAboveCurrentCountry);
+        push(firstBehindCurrentCountry, secondBehindCurrentCountry);
+        state.loadingMoreCountries = state.filteredCountry;
+      }
+    },
```

Comment:
Whole `getMoreOptions` function needs to be refactored

1. Too many magic numbers:
    You can define some constants and use `countriesInfo.length - 1` as number `249`. Same on `0`, `1` and `2`

![image](https://user-images.githubusercontent.com/42958020/183249859-0b1d9c08-a402-477e-90ef-4717a87a81b1.png)

2. `push` and `unshift` functions:
   Not necessary to put as a function, its just one line and the naming is not clear and too general. Also, state changes are hard to notice.

3. `state.loadingMoreCountries = state.filteredCountry;` can be extracted from if/else if. Its duplicated

4. a lot of `undefined` in the `filteredCountry` or `loadingMoreCountries` array, which means the application would crash